src/system.jl

```diff
@@ -0,0 +1,331 @@
+    pmin::KeyedArray{Float64, 2}
+    "Generator maximum output (MW)"
+    pmax::KeyedArray{Float64, 2}
+    "Ancillary services regulation offer prices (\$ /MW)"
+    asm_regulation::KeyedArray{Float64, 2}
+    "Ancillary services spinning offer prices (\$ /MW)"
+    asm_spin::KeyedArray{Float64, 2}
+    "Ancillary services supplemental on offer prices (\$ /MW)"
+    asm_sup_on::KeyedArray{Float64, 2}
+    "Ancillary services supplemental off offer prices (\$ /MW)"
+    asm_sup_off::KeyedArray{Float64, 2}
+
+    # Load time series
+    "Load time series data. `KeyedArray` where the axis keys are `load ids x datetimes`"
+    loads::KeyedArray{Float64, 2}
+
+    # Virtuals/PSD time series
+    "Increment bids time series data. `KeyedArray` where the axis keys are `bid ids x datetimes`"
+    increment_bids::KeyedArray{Vector{Tuple{Float64, Float64}}, 2}
+    "Decrement bids time series data. `KeyedArray` where the axis keys are `bid ids x datetimes`"
+    decrement_bids::KeyedArray{Vector{Tuple{Float64, Float64}}, 2}
+    "Price sensitive demand time series data. `KeyedArray` where the axis keys are `bid ids x datetimes`"
+    price_sensitive_demand::KeyedArray{Vector{Tuple{Float64, Float64}}, 2}
+end
+
+"""
+    $TYPEDEF
+
+Subtype of a `System` for modelling the real-time market.
+
+Fields:
+$TYPEDFIELDS
+"""
+struct SystemRT <: System
+    "`Dictionary` where the keys are bus names and the values are generator ids at that bus"
+    gens_per_bus::Dictionary{BusName, Vector{Int}}
+    "`Dictionary` where the keys are bus names and the values are load ids at that bus"
+    loads_per_bus::Dictionary{BusName, Vector{String}}
+
+    "Zones in the `System`, which will also include a `Zone` entry for the market wide zone"
+    zones::Dictionary{Int, Zone}
+    "Buses in the `System` indexed by bus name"
+    buses::Dictionary{BusName, Bus}
+    "Generators in the `System` indexed by unit code"
+    generators::Dictionary{Int, Generator}
+    "Branches in the `System` indexed by branch name"
+    branches::Dictionary{BranchName, Branch}
+    """
+    The line outage distribution factor matrix of the system for a set of contingencies given
+    by the keys of the `Dictionary`. Each entry is a `KeyedArray` with axis keys
+    `branch names x branch on outage`
+    """
+    LODF::Dictionary{String, KeyedArray{Float64, 2}}
+    """
+    Power transfer distribution factor of the system.  `KeyedArray` where the axis keys are
+    `branch names x bus names`
+    """
+    PTDF::KeyedArray{Float64, 2}
+
+    # Generator related time series
+    "Generation of the generator at the start of the time period (MW)"
+    initial_generation::KeyedArray{Float64, 1}
+    "Generator offer curves. `KeyedArray` where the axis keys are `generator names x datetimes`"
+    offer_curve::KeyedArray{Vector{Tuple{Float64, Float64}}, 2}
+    "Generator status indicated by a `Bool`"
+    status::KeyedArray{Bool, 2}
+    "Generator ancillary service status indicated by a `Bool`"
```

Comment:
```suggestion
    "Generator regulation status indicated by a `Bool`"
```
This is just for regulation, the status of other services is determined by `status` (i.e., the generator being committed or not)

Reply:
Also, I think we can probably improve these docstrings, right now they're basically just repeating the variable name and type (but not a priority, we can look into this later)